### PR TITLE
Fix for temp == 0 issue #684

### PR DESCRIPTION
**Pull request by Fabio3rs**
I am not sure about comparing the float directly with 0.0f, so I used std::abs and FLT_EPSILON

Motivation:
Setting temp=0 does not work as expected #684

### CODE REVIEW

File: llama.cpp [hunk 1]
```diff
@@ -1757,6 +1758,12 @@ llama_token llama_sample_top_p_top_k(
     // TODO: avoid this ...
     const auto last_n_tokens = std::vector<llama_token>(last_n_tokens_data, last_n_tokens_data + last_n_tokens_size);
 
+    if (std::abs(temp) < FLT_EPSILON) {
+        temp = 0.8f;
+        top_k = 1.0f;
```
Comment (abetlen) [inline llama.cpp:1763]: I believe this should be `top_k = 1;` as it's an integer.

Comment (Fabio3rs) [reply llama.cpp]: You are correct. Thanks.

**Review by abetlen — APPROVED** [review]

**Comment by abetlen** [comment]
Tested and it's working as expected now. Thank you!

**Comment by ivanstepanovftw** [comment]
Take a look at another approach https://github.com/ggerganov/llama.cpp/pull/720

**Comment by prusnak** [comment]
Closed in favor of https://github.com/ggerganov/llama.cpp/pull/720

**Pull request closed without merging**